crates/bin/prove_block/src/lib.rs

```diff
@@ -42,73 +43,20 @@ mod state_utils;
 mod types;
 mod utils;
 
-use std::fmt;
-
-// Define your error enum
-#[derive(Debug)]
+#[derive(Debug, Error)]
```

Comment:
Nitpick: there are so many `Error` traits, specifying that you're deriving `thiserror::Error` clarifies things.